docs/design/flow-traffic-mirroring/flow-traffic-mirroring.md

```diff
@@ -0,0 +1,182 @@
+# Flow traffic mirroring
+
+## Idea
+There must be an ability to mirror traffic on a given flow and send it to a specific switch+port+vlan.
+
+## Model
+FlowPath has a list of FlowMirrorPoints, which will contain MirrorGroupId for multicasting on the switch. 
+FlowMirrorPoint itself has a list of MirrorFlowPaths. MirrorFlowPaths contains the endpoint 
+where the traffic should be mirrored and other necessary information to build the path.
+
+![DB model](./model.png "DB model")
+
+If the receiver point is on the mirroring source switch, then no segments will be built for such a MirrorFlowPath, 
+and we will work with it like the paths in one switch flow.
+
+## API
+* Create mirror point:
+  
+  `PUT /flows/{flow_id}/mirror`
+  
+  payload:
+  ```
+  {
+      "flow_id": string,
+      "mirror_point": switch_id,
+      "mirror_direction": string [FORWARD|REVERSE],
+      "receiver_point": {
+         "switch_id": string,
+         "port": int,
+         "inner_vlan": int,
+         "outer_vlan": int
+      }
+  }
+  ```
+
+* Delete mirror point:
+
+  `DELETE /flows/{flow_id}/mirror`
+
+  payload:
+  ```
+  {
+      "flow_id": string,
+      "mirror_point": switch_id,
+      "mirror_direction": string [FORWARD|REVERSE],
+      "receiver_point": {
+          "switch_id": string,
+          "port": int,
+          "inner_vlan": int,
+          "outer_vlan": int
+      }
+  }
+  ```
+
+* Get mirror points:
+
+  `GET /flows/{flow_id}/mirror`
+
+  Response payload:
+  ```
+  {
+      "flow_id": string,
+      "points":[
+          {
+              "mirror_point": switch_id,
+              "mirror_direction": string [FORWARD|REVERSE],
+              "receiver_point": {
+                  "switch_id": string,
+                  "port": int,
+                  "inner_vlan": int,
+                  "outer_vlan": int
+              }
+          }
+      ]
+  }
+  ```
+
+
+* API that needs to be updated: 
+  - Need to add information about the state of the mirror paths in the flow payload.
+  - It is necessary to add information about built mirror paths to API `GET /flows/{flow_id}/path`
```

Comment:
Will it be V2 API? If so, please, rename the resource to /flows/{flow_id}/paths

Reply:
I believe that adding a new endpoint to the V2 API requires a separate design. And this task goes beyond the scope of the current task. Therefore, for now, information about new paths will be added to the already existing V1 API. Although I agree that this solution is not very good.